### ISSUE TITLE
[Bug]: Map layers (Feature, DEM Terrain) are not redrawn when toggling map style

### ISSUE THREAD

**Issue by Gnative** [rnmapbox/maps#2459]
### Mapbox Implementation

Mapbox

### Mapbox Version

default

### Platform

iOS (only)

### `@rnmapbox/maps` version

#main

### Standalone component to reproduce

```javascript
import React from 'react';
import { Button } from 'react-native';
import {
    MapView,
    Camera,
    ShapeSource,
    LineLayer,
} from '@rnmapbox/maps';

const MAP_STYLE_DEFAULT = 'mapbox://styles/mapbox/streets-v11'
const MAP_STYLE_OUTDOORS = 'mapbox://styles/mapbox/outdoors-v11'

class BugReportExample extends React.Component {

    state = {
        mapStyle: MAP_STYLE_DEFAULT,
        showLayer: false,
    };

    toggleMapStyle = ( ) => {
        this.setState({
            mapStyle: this.state.mapStyle === MAP_STYLE_DEFAULT ?  MAP_STYLE_OUTDOORS : MAP_STYLE_DEFAULT
        })
    }

    toggleLayer = ( ) => {
        this.setState({
            showLayer: !this.state.showLayer
        })
    }

    renderLayer = () => {
        if(!this.state.showLayer) return null

        return (<ShapeSource
                id="route"
                shape={{
                    id: 'route',
                    type: 'Feature',
                    geometry: {
                        type: 'LineString',
                        coordinates: [[10.141653, 47.158206],[10.141653, 47.265968], [10.33635, 47.265968], [10.33635,47.158206], [10.141653, 47.158206]],
                    },
                }
                }
            >
                <LineLayer
                    id="route-line"
                    style={{
                        lineColor: 'orange',
                        lineWidth: 10,
                    }}
                />
            </ShapeSource>
        )
    }

    render() {
        return (
            <>
                <Button title="Toggle Map Style" onPress={this.toggleMapStyle} />
                <Button title="Toggle Layer" onPress={this.toggleLayer} />
                <MapView
                    style={{ flex: 1 }}
                    styleURL={this.state.mapStyle}
                >
                    <Camera
                        animationDuration={0}
                        bounds={{
                            ne: [10.141653, 47.158206],
                            sw: [10.33635, 47.265968],
                        }}
                    />
                    {this.renderLayer()}
                </MapView>
            </>
        );
    }
}

export default BugReportExample;

```

### Observed behavior and steps to reproduce

The map layer is not being redrawn on map style change if it did not previously exist when the map was first created. 
This issue also applies if we were to toggle the 3D layer `<RasterDemSource> <Terrain />` in place of `<ShapeSource><LineLayer />`. 

This issue does not exist on Android

To replicate this issue in this example:

1. Click "Toggle Feature" to draw the map layer
2. Click "Toggle Map"

Then the layer is no longer showing and will throw subsequent errors such as `Mapbox error', 'removing layer failed for layer ...`

**Comment by BubbleTrouble14** [comment]
I have an issue when changing the styleURL it moves to camera back to its default position:
```
      <MapboxGL.MapView
        compassEnabled={true}
        compassPosition={Platform.OS === 'ios' ? { top: 276, right: 8 } : { bottom: 56, right: 4 }}
        scaleBarEnabled={false}
        style={{ flex: 1 }}
        styleURL={mapType}>
          <MapboxGL.Camera
            ref={camera}
            animationMode={'none'}
            followUserLocation={followLocation}
            followUserMode="normal"
            followZoomLevel={20}
            allowUpdates={true}
            defaultSettings={{
              bounds: {
                ne: [location.boundingBox.xmax, location.boundingBox.ymax],
                sw: [location.boundingBox.xmin, location.boundingBox.ymin],
              },
              padding: {
                paddingBottom: 40,
                paddingLeft: 40,
                paddingRight: 40,
                paddingTop: 40,
              },
            }}
          />
        </MapboxGL.MapView>
```

**Comment by FrancoisVA** [comment]
I get this same issue. All ShapeSources disappear after changing map style (styleURL) instead of re-rendering. This was an issue a couple of versions ago, then it was fixed, and now it's back. Did not keep track of which version.

**Comment by mfazekas** [comment]
Fixed by: #2488

**Comment by Gnative** [comment]
Perfect, Thanks @mfazekas
